File: programmers/체육복.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

int solution(int n, vector<int> lost, vector<int> reserve) {
    int answer = 0;
    
    for (int i = 0; i < lost.size(); i++)
    {
        for (int j = 0; j < reserve.size(); j++) 
        {
            if (lost[i] == reserve[j])
            {
                lost.erase(lost.begin() + i);
                reserve.erase(reserve.begin() + j);
                i--;
            }
        }
    }
    
    sort(lost.begin(), lost.end());
    sort(reserve.begin(), reserve.end());
    
    for (int j = 0; j < reserve.size(); j++)
    {
        for (int i = 0; i < lost.size(); i++)
        {
            if (lost[i] == reserve[j] - 1 || lost[i] == reserve[j] + 1)
            {
                lost.erase(lost.begin() + i);
                reserve.erase(reserve.begin() + j);
                j--;
            }
        }
    }
    
    answer = n - lost.size();
    
    return answer;
}

File: programmers/seohee/체육복.py
def solution(n, lost, reserve):
    lost.sort()
    reserve.sort()

    for i in lost[:]:
        if i in reserve:
            lost.pop(lost.index(i))
            reserve.pop(reserve.index(i))
    
    reserve_count = [0] * (n+2)
    
    num = n - len(lost)
    print(lost, reserve, num)
    for i in reserve:
        reserve_count[i] += 1
    
    for i in range(len(lost)):
        if reserve_count[lost[i]-1] != 0:
            num += 1
            reserve_count[lost[i]-1] = 0
        elif reserve_count[lost[i]+1] != 0:
            num += 1
            reserve_count[lost[i]+1] = 0
    return num